lib/auth.js

```diff
@@ -52,6 +52,26 @@ var auth = {
     } else {
       return cb(null, false);
     }
+  },
+  /**
+   * New endpoint required as autentication ($fh.auth) requests not handled
+   * correctly in local development/Studio Preview -
+   * the js sdk dertermines envrironment mode from url string.
+   * New endpoint determines environment mode from env var,
+   * sends request (req) and calls user provided function (cb).
+   * @param  {object} req
+   * @param  {object} res
+   * @param  {object} cb
+   */
+  performAuth: function (req, res, cb) {
```

Comment:
it might be better to make the `res` parameter optional. Make it clear that it is only used in local mode in the comments. 

Also I am not sure if it's enough to only support `res` as a JSONObject. Sometimes users might need more complicated authentication process. I suggest to make it a function instead (and perhaps call it "localAuth") to make it clear.

Then the implementation could be like this:

```
if(typeof localAuth === 'function') {
  return localAuth(req, cb);
} else if(typeof localAuth === 'object') {
  return cb(null, localAuth);
} else {
  return cb(null, {});
}
```